packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -285,11 +395,31 @@ export class TokenDetectionController extends StaticIntervalPollingController<
   }
 
   /**
-   * Triggers asset ERC20 token auto detection for each contract address in contract metadata on mainnet.
+   * Restart token detection polling period and call detectNewTokens
+   * in case of address change or user session initialization.
    *
-   * @param options - Options to detect tokens.
+   * @param options - Options for restart token detection.
+   * @param options.selectedAddress - the selectedAddress against which to detect for token balances
+   * @param options.chainId - the chainId against which to detect for token balances
+   */
+  async #restartTokenDetection({
+    selectedAddress,
+    chainId,
+  }: Partial<{ selectedAddress: string; chainId: Hex }> = {}) {
+    await this.detectTokens({
+      accountAddress: selectedAddress,
+      networkClientId: chainId,
```

Comment:
Hmm, the network client ID and chain ID aren't the same thing.

Can we have this method take a `networkClientId`?

Reply:
Fixed to align with `detectTokens` inputs: https://github.com/MetaMask/core/pull/3775/commits/e40e493d3aec680ad5b83f65885b25e4fe7b7b06